src/api/cpp/backend/backend_engine.h

```diff
@@ -19,169 +19,148 @@
+      return NIXL_SUCCESS;
+    } else {
+      return NIXL_ERR_NOT_ALLOWED;
+    }
+  }
+
+  nixl_status_t getInitParam(const std::string& key, std::string& value)
+  {
+    if (customParams->count(key) == 0) {
+      return NIXL_ERR_INVALID_PARAM;
+    } else {
+      value = (*customParams)[key];
+      return NIXL_SUCCESS;
+    }
+  }
+
+ public:
+  nixlBackendEngine(const nixlBackendInitParams* init_params) : localAgent(init_params->localAgent)
+  {
+    this->backendType = init_params->type;
+    this->initErr = false;
+    this->customParams = new nixl_b_params_t(*(init_params->customParams));
+  }
+
+  virtual ~nixlBackendEngine() { delete customParams; }
+
+  bool getInitErr() { return initErr; }
+  nixl_backend_t getType() const { return backendType; }
+  nixl_b_params_t getCustomParams() const { return *customParams; }
+
+  // The support function determine which methods are necessary by the child backend, and
+  // if they're called by mistake, they will return error if not implemented by backend.
+
+  // Determines if a backend supports remote operations
+  virtual bool supportsRemote() const = 0;
+
+  // Determines if a backend supports local operations
+  virtual bool supportsLocal() const = 0;
+
+  // Determines if a backend supports sending notifications. Related methods are not
+  // pure virtual, and return errors, as parent shouldn't call if supportsNotif is false.
+  virtual bool supportsNotif() const = 0;
+
+  // Determines if a backend supports progress thread.
+  virtual bool supportsProgTh() const = 0;
+
+  virtual nixl_mem_list_t getSupportedMems() const = 0;
+
+
+  // *** Pure virtual methods that need to be implemented by any backend *** //
+
+  // Register and deregister local memory
+  virtual nixl_status_t registerMem(const nixlBlobDesc& mem, const nixl_mem_t& nixl_mem, nixlBackendMD*& out) = 0;
+  virtual nixl_status_t deregisterMem(nixlBackendMD* meta) = 0;
+
+  // Make connection to a remote node identified by the name into loaded conn infos
+  // Child might just return 0, if making proactive connections are not necessary.
+  // An agent might need to connect to itself for local operations.
+  virtual nixl_status_t connect(const std::string& remote_agent) = 0;
+  virtual nixl_status_t disconnect(const std::string& remote_agent) = 0;
+
+  // Remove loaded local or remtoe metadata for target
+  virtual nixl_status_t unloadMD(nixlBackendMD* input) = 0;
+
+  // Preparing a request, which populates the async handle as desired
+  virtual nixl_status_t prepXfer(
+      const nixl_xfer_op_t& operation, const nixl_meta_dlist_t& local, const nixl_meta_dlist_t& remote,
+      const std::string& remote_agent, nixlBackendReqH*& handle, const nixl_opt_b_args_t* opt_args = nullptr) = 0;
+
+  // Posting a request, which completes the async handle creation and posts it
+  virtual nixl_status_t postXfer(
+      const nixl_xfer_op_t& operation, const nixl_meta_dlist_t& local, const nixl_meta_dlist_t& remote,
```

Comment:
IMO we had a better alignment here.
Is there a way to force custom indentation rules for specific places?